Onboarding Jack Dietrich

Jack-Dietrich opened the pull request:
# Purpose
Finished the onboarding challenge, looking for feedback.

# New Changes
Implemented the readTempLM75BD, Thermal Management Task, and osHandlerLM75BD

# Testing
Unit tests and Integration test ran.

# Outstanding Changes
N/A

Jack-Dietrich committed:
Initial Commit, rough draft of readTemp function.

Jack-Dietrich committed:
Successfull compilation. Not successfull conversion.

Jack-Dietrich committed:
LM75BD driver finished

Jack-Dietrich committed:
All unit tests passing for driver

Jack-Dietrich committed:
Begun osHandler work

Jack-Dietrich committed:
Fixed issues with OS handler

Jack-Dietrich committed:
Fixed buffer prolem in thermalMgr

Jack-Dietrich committed:
Code clean up

Jack-Dietrich committed:
Further clean up

readTempLM75BD function
--
-changed error catching functionality to use helper macros
-added nullptry check for dereferencing temp
-Added U unsigned prefix for address definition

OS handler
-moved temp read functionality out of this function to Thermal Mgr task

Thermal Mgr Task
-added new thermal manager event queue type that is sent by osHandler, recieved by thermalMgr.

kepler452b123 commented on lm75bd/lm75bd.c:
```diff
@@ -25,9 +25,60 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
+  
+
+
+  LOG_IF_ERROR_CODE(i2cReceiveFrom(LM75BD_OBC_I2C_ADDR, readBuff, 2));//read in 2 bytes back from the LM75BD
```
should return here as we don't want the function to continue if they fail

kepler452b123 commented on lm75bd/lm75bd.c:
```diff
@@ -25,9 +25,60 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
+  }
+
+  if(temp){//dereference only if not nullptr
+    *temp = convTemp;
```
we always check if a pointer param is null at the start of a function (and return invalid_arg if it is)

kepler452b123 commented on services/thermal_mgr/thermal_mgr.c:
```diff
@@ -44,18 +45,70 @@ void initThermalSystemManager(lm75bd_config_t *config) {
+
+          safeOperatingConditions();
+          
+        }
```
good, but implement logic to skip the rest of the logic if the driver function fails. e.g. check if errcode is success before continuing, or use continue statement if errCode is false. we don't want to do the rest of the logic if the driver function fails

Jack-Dietrich committed:
Fixes from pull request
-Changed logging if errer code to returning in the lm75bd driver function
-Moved pointer null check to start of lm75bd driver function
-added logic to check if temp read fails, if so skipping subsequent temp checks in thermal_mgr.c

kepler452b123 reviewed (commented):
lgtm send email to me
